core/modules/field/field.module

```diff
@@ -1308,6 +1309,72 @@ function field_get_block_list() {
   return $types;
 }
 
+/**
+ * Helper function to retrieve the most commonly used label for a field.
+ *
+ * @param string $machine_name
+ *   The machine name for the field in question.
+ *
+ * @return string
+ *   The most commonly used machine name.
+ */
+function field_get_label($machine_name) {
+  if (!isset($machine_name)) {
+    return FALSE;
+  }
+
+  $all_labels = field_get_labels($machine_name);
+  // Sort the labels from least common to most common.
+  asort($all_labels);
+  // Flip keys and values so array_pop can get us the key.
+  $labels_inverted = array_flip($all_labels);
+  // Pull the last (most commonly used) value.
+  $label = array_pop($labels_inverted);
+
+  return $label;
+}
+
+/**
+ * Helper function to retrieve all labels for a field.
+ *
+ * @param string $machine_name
+ *   The machine name for the field in question.
+ *
+ * @return array
+ *   An array with all the labels used for a field.
+ */
+function field_get_labels($machine_name = NULL) {
```

Comment:
Why should we allow `NULL` here?

Reply:
@klonos if machine_name is NULL then it returns all field labels.

Reply:
That should be put in the docblock documentation for $machine_name param and the `@return` documentation.

Or, unless we need it consider removing accepting NULL as a valid input. I don't like the pattern of returning two different array structures based on input. If a machine name is passed in, it's a simple array. If NULL is passed in, it's an array of arrays.

Additionally, return array needs to specify the structure of the array. Apparently it's an array where the key is the field label and the value is the number of times that label is used? Is that right?